src/main/scala/au/org/ala/biocache/index/IndexDAO.scala

```diff
@@ -816,317 +824,321 @@ trait IndexDAO {
-        i = i + 1
-        addField(doc,header(i), sensitiveMap.getOrElse("locality", ""))
-        i = i + 1
-        addField(doc,header(i), map.getOrElse("eventID", ""))
-        i = i + 1
-        addField(doc,header(i), map.getOrElse("locationID", ""))
-        i = i + 1
-        addField(doc,header(i), map.getOrElse("datasetName", ""))
-        i = i + 1
-        addField(doc,header(i), map.getOrElse("reproductiveCondition", ""))
-        i = i + 1
+            if (StringUtils.isNotEmpty(j)) addField(doc, header(i), j)
+        i += 1
+        addField(doc, header(i), sensitiveMap.getOrElse("coordinateUncertaintyInMeters.p", ""))
+        i += 1
+        addField(doc, header(i), distanceOutsideExpertRange)
+        i += 1
+        addField(doc, header(i), map.getOrElse("verbatimElevation.p", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("minimumElevationInMeters.p", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("maximumElevationInMeters.p", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("verbatimDepth.p", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("minimumDepthInMeters.p", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("maximumDepthInMeters.p", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("nameParseType.p", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("occurrenceStatus.p", "present"))
+        i += 1
+        addField(doc, header(i), map.getOrElse("occurrenceDetails", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("photographer", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("rights", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("georeferenceVerificationStatus", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("occurrenceStatus", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("locality", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("decimalLatitude", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("decimalLongitude", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("geodeticDatum", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("sex", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("lifeStage", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("behavior", ""))
+        i += 1
+        addField(doc, header(i), sensitiveMap.getOrElse("locality", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("eventID", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("locationID", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("datasetName", ""))
+        i += 1
+        addField(doc, header(i), map.getOrElse("reproductiveCondition", ""))
+        i += 1
```

Comment:
Hope you've enjoyed this method as much as I did :p

Reply:
Oh yes, but not enough to refactor it into loop